messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/token/store/TokenStore.java

```diff
@@ -66,22 +62,12 @@ public interface TokenStore {
      * @throws UnableToClaimTokenException When a segment has already been created.
      */
     @Nonnull
-    default CompletableFuture<Void> initializeTokenSegments(
+    CompletableFuture<List<Segment>> initializeTokenSegments(
```

Comment:
Very strictly speaking, this should probably be a `Set<Segment>` instead of a `List<Segment>`?